dynamic-layers/qt5-layer/recipes-qt/qt5/qtwebengine/0001-imxvpu-video-decode-accelerator.patch

```diff
@@ -0,0 +1,1884 @@
+ }
+ #endif
+ 
++#if defined(IMX_PLATFORM)
++std::unique_ptr<VideoDecodeAccelerator>
++GpuVideoDecodeAcceleratorFactory::CreateImxVpuVDA(
++    const gpu::GpuDriverBugWorkarounds& workarounds,
++    const gpu::GpuPreferences& gpu_preferences) const {
++        VLOG(3) << "GpuVideoDecodeAcceleratorFactory::CreateImxVpuVDA";
++  std::unique_ptr<VideoDecodeAccelerator> decoder;
++  decoder.reset(new ImxVpuVideoDecodeAccelerator(
++        get_context_group_cb_, make_context_current_cb_));
++  return decoder;
++}
++#endif
++
+ GpuVideoDecodeAcceleratorFactory::GpuVideoDecodeAcceleratorFactory(
+     const GetGLContextCallback& get_gl_context_cb,
+     const MakeGLContextCurrentCallback& make_context_current_cb,
+diff --git a/chromium/media/gpu/gpu_video_decode_accelerator_factory.h b/chromium/media/gpu/gpu_video_decode_accelerator_factory.h
+index 56a0d7fd16..b6d6152d4d 100644
+--- a/chromium/media/gpu/gpu_video_decode_accelerator_factory.h
++++ b/chromium/media/gpu/gpu_video_decode_accelerator_factory.h
+@@ -121,6 +121,11 @@ class MEDIA_GPU_EXPORT GpuVideoDecodeAcceleratorFactory {
+       const gpu::GpuDriverBugWorkarounds& workarounds,
+       const gpu::GpuPreferences& gpu_preferences) const;
+ #endif
++#if defined(IMX_PLATFORM)
++  std::unique_ptr<VideoDecodeAccelerator> CreateImxVpuVDA(
++      const gpu::GpuDriverBugWorkarounds& workarounds,
++      const gpu::GpuPreferences& gpu_preferences) const;
++#endif
+ 
+   const GetGLContextCallback get_gl_context_cb_;
+   const MakeGLContextCurrentCallback make_context_current_cb_;
+diff --git a/chromium/media/gpu/imx/imx_gl_viv_direct_texture.cc b/chromium/media/gpu/imx/imx_gl_viv_direct_texture.cc
+new file mode 100644
+index 0000000000..f03902d1f8
+--- /dev/null
++++ b/chromium/media/gpu/imx/imx_gl_viv_direct_texture.cc
+@@ -0,0 +1,38 @@
++#include "ui/gl/gl_implementation.h"
++#include "imx_gl_viv_direct_texture.h"
++
++
++bool init_viv_direct_texture(GLESVIVDirectTextureProcs &procs)
++{
++  VLOG(1) << "Initializing Vivante direct texture GLES extension";
++
++  gl::GLImplementation glimpl = gl::GetGLImplementation();
++  if (glimpl != gl::kGLImplementationEGLGLES2)
++  {
++    LOG(INFO) << "Cannot initialize direct textures - GL implementation is "
++              << gl::GetGLImplementationName(glimpl)
++              << ", expected " <<
++              gl::GetGLImplementationName(gl::kGLImplementationEGLGLES2);
++    return false;
++  }
++
++  // Newer Vivante drivers call the extension GL_VIV_tex_direct instead of GL_VIV_direct_texture,
++  // even though it is the same extension
++  // if (context.HasExtension("GL_VIV_direct_texture"))
++  //   VLOG(1) << "GL_VIV_direct_texture supported";
++  // else if (context.HasExtension("GL_VIV_tex_direct"))
++  //   VLOG(1) << "GL_VIV_tex_direct supported";
++  // else
++  // {
++  //   VLOG(1) << "Neither GL_VIV_direct_texture nor GL_VIV_tex_direct supported";
++  //   return false;
++  // }
```

Comment:
Shouldn't this code be enabled?

Reply:
I didn't know how to get the context here from ImxVpuVideoDecodeAccelerator::Initialize. That's why I commented out this part. As usual, any suggestions welcome.